utils/collection/search.go

```diff
@@ -71,6 +71,44 @@ func AnyFunc[S ~[]E, E any](s S, f func(E) bool) bool {
 	return conditions.Any()
 }
 
+// Filter returns a new slice that contains elements from the input slice which return true when they’re passed as a parameter to the provided filtering function f.
+func Filter[S ~[]E, E any](s S, f func(E) bool) (result S) {
+	result = make(S, 0, len(s))
+
+	for i := range s {
+		if f(s[i]) {
+			result = append(result, s[i])
+		}
+	}
+
+	return result
+}
+
+// Map creates a new slice and populates it with the results of calling the provided function on every element in input slice.
+func Map[T1 any, T2 any](s []T1, f func(T1) T2) (result []T2) {
+	result = make([]T2, len(s))
+
+	for i := range s {
+		result[i] = f(s[i])
+	}
+
+	return result
+}
+
+// Reject is the opposite of Filter and returns the elements of collection for which the filtering function f returns false.
```

Comment:
Useful to remind of how it is different to DeleteFunc 

```suggestion
// Reject is the opposite of Filter and returns the elements of collection for which the filtering function f returns false.
// This is functionally equivalent to slices.DeleteFunc but it returns a new slice.
```